source/codegen/templates/service_helpers.mako

```diff
@@ -420,14 +432,39 @@ ${initialize_standard_input_param(function_name, parameter)}
 % endif
 </%def>
 
+<%def name="set_output_vararg_parameter_sizes(parameters)">\
+<%
+  input_vararg_parameter = [p for p in parameters if common_helpers.is_repeated_varargs_parameter(p) and common_helpers.is_input_parameter(p)][0]
+  input_parameter_name = common_helpers.camel_to_snake(input_vararg_parameter['cppName'])
+  output_vararg_parameters = [p for p in parameters if common_helpers.is_repeating_parameter(p) and not common_helpers.is_input_parameter(p)]
+%>\
+% for parameter in output_vararg_parameters:
+<%
+  parameter_name = common_helpers.camel_to_snake(parameter['cppName'])
+%>\
+      ${parameter_name}Vector.resize(${input_parameter_name}.size());
+% endfor
+</%def>
+
 ## Initialize the output parameters for an API call.
 <%def name="initialize_output_params(output_parameters)">\
 % for parameter in output_parameters:
 <%
   parameter_name = common_helpers.camel_to_snake(parameter['cppName'])
   underlying_param_type = common_helpers.get_underlying_type_name(parameter["type"])
 %>\
-%   if common_helpers.is_array(parameter['type']):
+%   if common_helpers.is_repeating_parameter(parameter):
+      auto get_${parameter_name}_if = [](std::vector<${underlying_param_type}>& vector, int n) -> ${underlying_param_type}* {
+            if (vector.size() > n) {
+## Note that this code will not handle every datatype, but it works for all
+## the ones we currently use with repeated varargs.
+                  return &(vector[n]);
+            }
+            return nullptr;
```

Comment:
Is there a good place to document that terminating with a nullptr is part of the convention and required (hence the extra pair of args)? Maybe around the this get_if code that implements it? Or possibly around the core/common helpers describing the mechanism?

It's OK if it's not obvious to a casual reader. As long as someone who really needs to figure it out has something to hang on to.

Reply:
I did document it in `create_args_for_varargs()` in `source/codegen/service_helpers.py` - I think that's good enough; let me know if you agree.